### PR TITLE
Dependency upgrade

### PR DESCRIPTION
**Pull request by simonmarty**
*Issue #, if available:*

*Description of changes:* Upgrade fast-xml-parser and GH action, fix IDE warnings.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: __tests__/utils.test.ts [hunk 1]
```diff
@@ -87,7 +84,7 @@ describe('Test secret value retrieval', () => {
     });
 
     test('Throws an error if unable to retrieve the secret', async () => {
-        const error = new ResourceNotFoundException({$metadata: {} as any, Message: 'Error'});
+        const error = new ResourceNotFoundException({$metadata: {}, message: 'Error'});
```
Comment (adithyasolai) [inline __tests__/utils.test.ts:87]: So are changes like this done by some linter, or done by you?

Is this just scope creep?

Comment (simonmarty) [reply __tests__/utils.test.ts:87]: Fixes build failure: https://github.com/aws-actions/aws-secretsmanager-get-secrets/actions/runs/5308032009/jobs/9607113113

**Review by adithyasolai — APPROVED** [review]

**Review by danmancuso — APPROVED** [review]

**Review by adithyasolai — APPROVED** [review]
